Sources/SwiftIfConfig/ActiveSyntaxRewriter.swift

```diff
@@ -307,3 +327,104 @@ class ActiveSyntaxRewriter<Configuration: BuildConfiguration>: SyntaxRewriter {
     return visit(rewrittenNode)
   }
 }
+
+/// Helper class to find a feature or compiler check.
+fileprivate class FindFeatureCheckVisitor: SyntaxVisitor {
+  var foundFeatureCheck = false
+
+  override func visit(_ node: DeclReferenceExprSyntax) -> SyntaxVisitorContinueKind {
+    // Checks that start with $ are feature checks that should be retained.
+    if let identifier = node.simpleIdentifier,
+      let initialChar = identifier.name.first,
+      initialChar == "$"
+    {
+      foundFeatureCheck = true
+      return .skipChildren
+    }
+
+    return .visitChildren
+  }
+
+  override func visit(_ node: FunctionCallExprSyntax) -> SyntaxVisitorContinueKind {
+    if let calleeDeclRef = node.calledExpression.as(DeclReferenceExprSyntax.self),
+      let calleeName = calleeDeclRef.simpleIdentifier?.name,
+      (calleeName == "compiler" || calleeName == "_compiler_version")
+    {
+      foundFeatureCheck = true
+    }
+
+    return .skipChildren
+  }
+}
+
+extension ExprSyntaxProtocol {
+  /// Whether any of the nodes in this expression involve compiler or feature
+  /// checks.
+  fileprivate var containsFeatureCheck: Bool {
+    let visitor = FindFeatureCheckVisitor(viewMode: .fixedUp)
+    visitor.walk(self)
+    return visitor.foundFeatureCheck
+  }
+}
+
+extension IfConfigDeclSyntax {
+  /// Whether any of the clauses in this #if contain a feature check.
+  var containsFeatureCheck: Bool {
+    return clauses.contains { clause in
+      if let condition = clause.condition {
+        return condition.containsFeatureCheck
+      } else {
+        return false
+      }
+    }
+  }
+}
+
+extension SyntaxProtocol {
+  // Produce the source code for this syntax node with all of the comments
+  // and #sourceLocations removed. Each comment will be replaced with either
+  // a newline or a space, depending on whether the comment involved a newline.
+  @_spi(Compiler)
+  public var descriptionWithoutCommentsAndSourceLocations: String {
+    var result = ""
+    var skipUntilRParen = false
+    for token in tokens(viewMode: .sourceAccurate) {
```

Comment:
`SyntaxVisitor` is _probably_ faster because it uses `SyntaxNodeFactory` hack.
```swift
class DescriptionWithoutCommentsAndSourceLocationsVisotr: SyntaxVisitor {
  var result: String = ""
  override func visit(_ token: TokenSyntax) -> SyntaxVisitorContinueKind {
    token.leadingTrivia.writeWithoutComments(to: &result)
    token.text.write(to: &result)
    token.trailingTrivia.writeWithoutComments(to: &result)
    return .skipChildren
  }
  override func visit(_ node: PoundSourceLocationSyntax) -> SyntaxVisitorContinueKind {
    return .skipChildren
  }
}
```

Reply:
This is much cleaner than my implementation, too!